### PR TITLE
Docstring and style improvements

### PR DESCRIPTION
**Pull request by brickZA**
Also added some TODO comments with performance improvement ideas for
inspecting_client.py

### CODE REVIEW

File: katcp/inspecting_client.py [hunk 1]
```diff
@@ -519,13 +519,38 @@ def handle_sensor_value(self):
...
-        """Inspect device requests and sensors, update model"""
+        """Inspect device requests and sensors, update model
+
+        Returns
+        -------
+
+        Tornado future that resolves with:
+
+        model_changes : Nested AttrDict
+            Contanins sets of added/removed request/sensor names
```
Comment (ajoubertza) [inline katcp/inspecting_client.py]: - [ ] Typo:  Contanins -> Contains

File: katcp/inspecting_client.py [hunk 2]
```diff
@@ -552,7 +577,20 @@ def inspect_requests(self, name=None, timeout=None):
         timeout : float or None, optional
             Timeout for request inspection, None for no timeout
 
-        TODO Return value
+        Returns
+        -------
+        Tornado future that resolves with:
+
+        request_changes : AttrDict of sets or None
+            Contanins sets of added/removed request names
```
Comment (ajoubertza) [inline katcp/inspecting_client.py]: - [ ] Typo:  Contanins -> Contains

File: katcp/inspecting_client.py [hunk 3]
```diff
@@ -519,13 +519,38 @@ def handle_sensor_value(self):
...
+            {'requests': {
+                'added': set(['req1', 'req2']),
+                'removed': set(['req10', 'req20'])}
+             'sensors': {
+                'added': set(['sens1', 'sens2']),
+                'removed': set(['sens10', 'sens20'])}
+            }
+
+            If there are no changes keys may be omitted. If an item is in both
+            the 'added' and 'removed' sets that means that it changed.
```
Comment (ajoubertza) [inline katcp/inspecting_client.py]: This comment sounds like you will get an empty dict if there were no changes at all.  
Suggest something like:

> - If an item is in both the 'added' and 'removed' sets that means that it changed.
> - If some, but not all types of changes occurred, keys may be omitted for 
> items that did not change.
> - If there are neither request nor sensor changes, resolves with None

File: katcp/inspecting_client.py [hunk 4]
```diff
@@ -519,13 +519,38 @@ def handle_sensor_value(self):
...
     def inspect(self):
-        """Inspect device requests and sensors, update model"""
+        """Inspect device requests and sensors, update model
+
+        Returns
+        -------
+
+        Tornado future that resolves with:
+
+        model_changes : Nested AttrDict
```
Comment (ajoubertza) [inline katcp/inspecting_client.py]: If there were neither request changes nor sensor changes, then the return value will resolve to `None`.
Suggest adding "or None".

File: katcp/inspecting_client.py [hunk 5]
```diff
@@ -593,7 +631,19 @@ def inspect_sensors(self, name=None, timeout=None):
         timeout : float or None, optional
             Timeout for sensors inspection, None for no timeout
 
-        TODO Return value
+        Returns
+        -------
+        Tornado future that resolves with:
+
+        request_changes : AttrDict of sets or None
+            Contanins sets of added/removed request names
```
Comment (ajoubertza) [inline katcp/inspecting_client.py]: - [ ] Typo:  request_changes -> sensor_changes
- [ ] Typo:  Contanins -> Contains
- [ ] Typo:  request -> sensor

**Comment by lvdheever** [comment]
@brickZA Please pick this PR up again.

File: katcp/inspecting_client.py (hunk 4 above)
Comment (brickZA) [inline katcp/inspecting_client.py]: Added in commit to follow.

File: katcp/inspecting_client.py (hunk 1 above)
Comment (brickZA) [inline katcp/inspecting_client.py]: Updated in commit to follow.

File: katcp/inspecting_client.py (hunk 3 above)
Comment (brickZA) [inline katcp/inspecting_client.py]: Updated in commit to follow.

File: katcp/inspecting_client.py (hunk 2 above)
Comment (brickZA) [inline katcp/inspecting_client.py]: Fixed in commit to follow.

File: katcp/inspecting_client.py (hunk 5 above)
Comment (brickZA) [inline katcp/inspecting_client.py]: Updated in commit to follow. The text was partially replaced by docstrings written by @bmerry in PR #176

**Comment by brickZA** [comment]
@ajoubertza Please see updates in most recent commit.

**Pull request merged**